### PR TITLE
Fixed 'No valid command: zero.' error

### PR DESCRIPTION
**Pull request by rapperskull**
An error in the logic considered invalid all lines NOT starting with a digit, instead of the ones starting with digits.

### CODE REVIEW

**Comment by rapperskull** [comment]
Sorry, I misunderstood that line. What gave me errors was not that check, but a new command, not handled by your script: **zero**.
Changing that line (because of a misunderstood comment), produced a working image file. I don't know if it's somehow corrupted, but ex4unpacker was able to extract it and text files are not corrupted.
So, even if this pull request is wrong, there's something different to handle in new Marshmallow scripts.

**Comment by xpirt** [comment]
Currently Marshmallow is not supported.
Would you like to uplaod your system.transfer.list? I'll check and add support as soon as possible.

**Comment by rapperskull** [comment]
Sure, here you go: https://gist.github.com/rapperskull/a38cb8ece695372fe1ff

**Comment by xpirt** [comment]
sdat2img.py: added marshmallow 6.0 support - d7211b1a35564ce612be8fa8699a2be701cde218

fixed.

**Comment by rapperskull** [comment]
Thank you, so the new information about zero blocks is not relevant?

**Comment by xpirt** [comment]
Of course it is! But not for the decompression process. 

The `zero` command was introduced as an workaround for bug 20881595 on google's source code which caused `dm-verity` (kernel driver that verifies integrity of a partition using a hash tree and signed metadata) to fail on some blocks. This is the commit which fixes it: https://github.com/android/platform_build/commit/2fd2c9bfd6c9beeaeccb4632b785227fe56c6006

Basically the `zero` command zeroes out those blocks which are marked as extended blocks to avoid dm-verity to fail. It is only needed when extracting blocks into partitions e.g when compressing into .dat file.

**Comment by rapperskull** [comment]
I thought about it: simply writing the other blocks does not assure that the blocks that should be zero are really zero, but since we're writing to a new file, and not to a real partition it doesn't matter.
